src/attributes/classes.js

```diff
@@ -24,19 +22,17 @@ jQuery.fn.extend({
 			});
 		}
 
-		if ( proceed ) {
-			// The disjunction here is for better compressibility (see removeClass)
-			classes = ( value || "" ).match( rnotwhite ) || [];
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+			while ( ( elem = this[ i++ ] ) ) {
```

Comment:
This is technically a behavior change, albeit one I'm in favor of. Still, we should probably document that collections including falsy elements are invalid context for (all?) jQuery methods. Would anyone in @jquery/core oppose?